### ISSUE TITLE
MapAttribute with a attribute constructor (null=True)

### ISSUE THREAD

**Issue by 3dw1np** [pynamodb/PynamoDB#309]
```
class Insurrance(MapAttribute):
      details = UnicodeAttribute(null=True)

class RentalModel(Model):
  ...
      Insurrance = Insurrance()
```

I have the following exception even if I specify null=True on the attribute constructor.

**Failed to put item: 
Parameter validation failed: Invalid type for parameter Item.insurrance.M.details.S, value: None, type: <type 'NoneType'>, valid types: <type 'basestring'>**

**Comment by jpinner-lyft** [comment]
Having difficulty reproducing -- could you provide a few more details? Perhaps a more complete stack trace or the `__dict__` fields for the model and map attributes before calling `save()`.

**Comment by 3dw1np** [comment]
@jpinner-lyft I think it was because I set details to None, instead of passing nothing.
Maybe it can be an improvement to support None value as null in dynamodb ?

**Comment by jpinner-lyft** [comment]
@3dw1np thanks, was able to reproduce by setting details to `None`. In particular it looks like we handle None strings differently in MapAttributes vs Models. As the example below shows, serialization ignores `NULL` values when they are top level attributes, but not when they are embedded, which is definitely a bug.
```
>>> class Insurrance(MapAttribute):
...     details = UnicodeAttribute(null=True)
>>> class RentalModel(Model):
...     id = UnicodeAttribute(hash_key=True)
...     note = UnicodeAttribute(null=True)
...     insurrance = Insurrance()
>>> rental_model = RentalModel(id='foo')
>>> rental_model.insurrance = {}
>>> rental_model.insurrance.details = None
>>> rental_model.note = None
>>> rental_model._serialize(attr_map=True)
{'attributes': {'id': {'S': u'bar'}, 'insurrance': {'M': {'details': {'S': None}}}}}
```

**Comment by jpinner-lyft** [comment]
Fixed in #375
